dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/InMemoryPermissionsImpl.java

```diff
@@ -0,0 +1,157 @@
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.DockstoreWebserviceConfiguration;
+import io.dockstore.webservice.core.Entry;
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.core.Workflow;
+import io.dockstore.webservice.core.WorkflowMode;
+import org.apache.http.HttpStatus;
+
+/**
+ * Implementation of the {@link PermissionsInterface} that only reads
+ * and saves settings in memory. When you bring down the
+ * Java VM, all settings are lost. If more than one instance of the
+ * Dockstore webservice were running, settings would NOT be replicated
+ * among the instances.
+ *
+ * <p>This is useful for testing without having any dependencies on an external
+ * authorization service.</p>
+ *
+ * <p>Note that this PermissionsInterface implementation does not support user groups.</p>
+ */
+public class InMemoryPermissionsImpl implements PermissionsInterface {
+
+    private final Map<String, Map<String, Role>> map = new ConcurrentHashMap<>();
+    private DockstoreWebserviceConfiguration configuration;
+
+    @Override
+    public List<Permission> setPermission(Workflow workflow, User requester, Permission permission) {
+        Map<String, Role> entryMap = map.get(workflow.getWorkflowPath());
+        if (entryMap == null) {
+            entryMap = new ConcurrentHashMap<>();
+            map.put(workflow.getWorkflowPath(), entryMap);
+        }
+        entryMap.put(permission.getEmail(), permission.getRole());
+        return getPermissionsForWorkflow(requester, workflow);
+    }
+
+    @Override
+    public List<String> workflowsSharedWithUser(User user) {
+        return map.entrySet().stream()
+                .filter(e ->  e.getValue().containsKey(user.getEmail()))
+                .map(e -> e.getKey())
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<Permission> getPermissionsForWorkflow(User user, Workflow workflow) {
+        List<Permission> owners = getHostedOwners(workflow);
+        Map<String, Role> permissionMap = map.get(workflow.getWorkflowPath());
+        if (permissionMap == null) {
+            return owners;
+        }
+        List<Permission> permissionList = permissionMap.entrySet().stream().map(e -> {
+            Permission permission = new Permission();
+            permission.setEmail(e.getKey());
+            permission.setRole(e.getValue());
+            return permission;
+        }).collect(Collectors.toList());
+        permissionList.addAll(owners);
+        return permissionList;
+    }
+
+    private List<Permission> getHostedOwners(Entry entry) {
+        List<Permission> list = new ArrayList<>();
+        if (entry instanceof Workflow) {
+            Workflow workflow = (Workflow)entry;
+            if (workflow.getMode() == WorkflowMode.HOSTED) {
```

Comment:
Its not clear to me, why are hosted workflows treated specially here? 
My guess: we were planning on testing out permissions with hosted workflows first. But it makes more sense to me to disable the GUI for permissions on anything but hosted workflows rather than treat it as a special case in the webservice. Could be misinterpreting. 

If it is necessary to write temp code that will be removed later, add a `//DODO` comment explaining

Reply:
I will remove this section. This class was lagging; I had started with it and was playing with it before I got SAM running locally, then when I got SAM working I stopped working on this.

It needs to be correct for testing, so I will go through it again.